src/main/java/org/skriptlang/skript/elements/fireworks/ExprAttachedTo.java

```diff
@@ -0,0 +1,101 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.elements.fireworks;
+
+import org.bukkit.entity.Firework;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Firework Attached To")
+@Description("Returns the entity attached to the fireworks if any.")
+@Examples({
+	"on firework explode:",
+		"\tloop all players in radius 10 around event-location:",
+			"\t\tloop-player is not attached entity of event-firework",
+			"\t\tpush loop-player upwards at speed 2"
+})
+@RequiredPlugins("Spigot 1.19.4+ or Paper")
```

Comment:
Is it 1.19.4+ for bith spigot and paper? if so use `Paper/Spigot 1.19.4+` unles it's Paper 1.12+ then it's fine otherwise define Paper's version

Reply:
It can be for any paper version or you need Spigot 1.19.4+.